src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -134,5 +127,268 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
+                var partsBuilder = ArrayBuilder<BoundExpression>.GetInstance(builderArguments.Length);
+                int baseStringLength = 0;
+                int numFormatHoles = 0;
+
+                for (int i = 0; i < builderArguments.Length; i++)
+                {
+                    var currentFormatCall = builderArguments[i];
+                    Debug.Assert(currentFormatCall.Arguments.Length > 0);
+                    Debug.Assert(argumentsBuilder.Count == 0);
+
+                    var newPart = currentFormatCall.Arguments[0];
+                    argumentsBuilder.Add(new BoundInterpolatedStringElementPlaceholder(newPart.Syntax, i, newPart.Type));
+                    if (currentFormatCall.Arguments.Length > 1)
+                    {
+                        argumentsBuilder.AddRange(currentFormatCall.Arguments, 1, currentFormatCall.Arguments.Length - 1);
+                    }
+
+                    interpolatedDataBuilder.Add(currentFormatCall with { Arguments = argumentsBuilder.ToImmutableAndClear() });
+
+                    var currentPart = unconvertedInterpolatedString.Parts[i];
+                    if (currentPart is BoundStringInsert insert)
+                    {
+                        numFormatHoles++;
+
+                        if (needToCheckConversionToObject)
+                        {
+                            Debug.Assert(conversionDiagnostics is not null);
+                            var value = insert.Value;
+                            bool reported = false;
+                            if (value.Type is not null)
+                            {
+                                value = BindToNaturalType(value, conversionDiagnostics);
+                                if (conversionDiagnostics.HasAnyErrors())
+                                {
+                                    CheckFeatureAvailability(value.Syntax, MessageID.IDS_FeatureImprovedInterpolatedStrings, diagnostics);
+                                    conversionDiagnostics.Clear();
+                                    reported = true;
+                                }
+                            }
+
+                            if (!reported)
+                            {
+                                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(conversionDiagnostics);
+                                var conversion = Conversions.ClassifyConversionFromExpression(value, objectType, ref useSiteInfo);
+                                if (!conversion.Exists || !conversion.IsValid || useSiteInfo.HasErrors)
+                                {
+                                    CheckFeatureAvailability(value.Syntax, MessageID.IDS_FeatureImprovedInterpolatedStrings, diagnostics);
+                                }
+                            }
+                        }
+
+                        newPart = insert.Update(newPart, insert.Alignment, insert.Format, newPart.Type);
```

Comment:
>newPart.Type [](start = 90, length = 12)

This looks suspicious because from the current sources it appears that ```BoundStringInsert.Type``` is always supposed to be null.